api/v1beta2/fakekindwithinfrastructuretemplate_types.go

```diff
@@ -0,0 +1,53 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta2
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+//+kubebuilder:object:root=true
+//+kubebuilder:storageversion
+
+// FakeKindWithInfrastructure is the Schema for testing CRDs with infrastructureTemplate under spec
+type FakeKindWithInfrastructureTemplate struct {
```

Comment:
why is this defined in the CAPC public API? I feel like it should be defined in a test package and not exposed to CAPC users.

Reply:
I agree you on this. still trying to figure out way to put in a test package, it is harder than expected.
The challenge here: 

1. a CRD needs to be created so that our envtest can load it.
2. a scheme needs to addToScheme so that we have the map between GVK and go struct

Currently, our test package has not set it up that a deep_copy and scheme builder markers and so on, eventually, test will looks like a API. Still not certain if this is the correct path.

Reply:
I managed to move it to dummies package by solved above mentioned challenge.

Reply:
Testing CAPC is fun huh?

Reply:
I felt your pain.